### PR TITLE
SDK Workflow

### PR DESCRIPTION
**Pull request by colin-axner**
Added build automation, will add replace clause into go.mod

Example: 
`make dev-local local=~/go/src/github.com/colin-axner`
use the local variable to specify the path to your fork

OR 

add sdk as a remote then:
```
git fetch sdk/master
make dev-sdk branch=master
```
replace "master" with desired branch. You still need to run make build/install. 

`make dev-clean` will remove any of the replace clauses added

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [x] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry: `clog add [section] [stanza] [message]`
- [x] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by alessio** [comment]
```
$ make dev-sdk
fatal: ambiguous argument 'sdk/': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
fatal: ambiguous argument 'sdk/': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
correct usage: make dev-sdk branch=[Branch on SDK Repo]
```

The bash snippet needs some care, I can help out with that

File: Makefile [hunk 1]
```diff
@@ -108,6 +108,18 @@ clean:
 distclean: clean
 	rm -rf vendor/
 
+dev-sdk:
+	@if [ -z "$(branch)" ]; then echo $ correct usage: make dev-sdk branch=[Branch on SDK Repo]; \
+	else hash='$(shell git rev-parse sdk/$(branch))'; time='$(shell git show -s sdk/$(branch) --date=format:'%Y%m%d%H%M%S' --format=%cd)'; \
+	$ sed -i '$$ a\replace github.com\/cosmos\/cosmos-sdk => github.com\/cosmos\/cosmos-sdk v0.0.0-'$$time'-'$$hash'' go.mod; fi
```
Comment (alessio) [inline Makefile]: The dollar sign at she beginning of the line should be removed

File: Makefile [hunk 2]
```diff
@@ -108,6 +108,18 @@ clean:
 distclean: clean
 	rm -rf vendor/
 
+dev-sdk:
+	@if [ -z "$(branch)" ]; then echo $ correct usage: make dev-sdk branch=[Branch on SDK Repo]; \
```
Comment (alessio) [inline Makefile]: I would simplify this

File: Makefile [hunk 3]
```diff
@@ -108,6 +108,18 @@ clean:
...
+	else hash='$(shell git rev-parse sdk/$(branch))'; time='$(shell git show -s sdk/$(branch) --date=format:'%Y%m%d%H%M%S' --format=%cd)'; \
```
Comment (alessio) [inline Makefile]: These instructions get executed anyway - this is make, not pure bash

**Review by alexanderbez — COMMENTED** [review]
Thanks for this @colin-axner! But I question if we actually need this technical overhead. As simple as it is to run this command it is equally as simple to just drop in a `replace` in the `go.mod` file -- i.e. I don't think we actually need this, but perhaps rather a quick small doc on how the process goes for development.

**Comment by alessio** [comment]
I agree with @alexanderbez here - docs is most valuable thing for users here

**Review by alexanderbez — APPROVED** [review]
ACK -- thank you

**Pull request merged**